human-forecasts/performance-board/app.R

```diff
@@ -0,0 +1,180 @@
+library(shiny)
+library(magrittr)
+library(ggplot2)
+library(scoringutils)
+
+
+forecast_files <- list.files("processed-forecast-data/")
+file_paths <- paste0("processed-forecast-data/", forecast_files)
+
+forecasts <- lapply(file_paths, readr::read_csv) %>%
+    dplyr::bind_rows() %>%
+    dplyr::filter(inc == "incident", 
+                  type == "quantile") %>%
+    dplyr::rename(prediction = value) %>%
+    dplyr::mutate(target_end_date = as.Date(target_end_date),
+                  forecast_date = as.Date(target_end_date), 
+                  type = ifelse(grepl("case", target), "cases", "deaths")) 
+
+deaths_inc <- data.table::fread(here::here("data", "weekly-incident-deaths.csv")) %>%
+    dplyr::mutate(inc = "incident", 
+                  type = "deaths")
+
+cases_inc <- data.table::fread(here::here("data", "weekly-incident-cases.csv")) %>%
+    dplyr::mutate(inc = "incident", 
+                  type = "cases")
+
+
+obs <- dplyr::bind_rows(deaths_inc, cases_inc) %>%
+    dplyr::rename(true_value = value) %>%
+    dplyr::mutate(target_end_date = as.Date(target_end_date)) %>%
+    dplyr::filter(epiweek >= lubridate::epiweek(Sys.Date()) - 10)
+
+combined <- dplyr::inner_join(obs, forecasts)
+
+combined_full <- dplyr::left_join(obs, forecasts)
```

Comment:
Usually safer to explicitly list the joining variables. Not doing this can lead to unexpected bugs down the line if input data changes.